### PR TITLE
Replaced Filterable with FilterFunc, also, some changes in the registry.go

### PR DESCRIPTION
**Pull request by zuzzas**
#### Overview

1. Replaced Filterable with FilterFunc
2. Removed Register(), left only RegisterFunc()
3. Added helper function to convert to and from Unstructured objects

#### What this PR does / why we need it

The Filterable interface is unwieldy and impossible to reason about. Replaced it with a simple function.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: sdk/registry.go [hunk 1]
```diff
@@ -22,9 +22,7 @@ var moduleRe = regexp.MustCompile(`/modules/(([^/]+)/hooks/([^/]+/)*([^/]+))$`)
 // TODO: This regexp should be changed. We shouldn't force users to name modules with a number prefix.
 var moduleNameRe = regexp.MustCompile(`^[0-9][0-9][0-9]-(.*)$`)
 
-var _ = initRegistry()
-
-func initRegistry() bool {
+var _ = func() struct{} {
```
Comment (diafour) [inline sdk/registry.go:25]: This initialization function and stubs below are redundant now.

**Review by diafour — CHANGES_REQUESTED** [review]

File: pkg/module_manager/global_hook_config.go [hunk 2]
```diff
@@ -298,12 +298,12 @@ func NewHookConfigFromGoConfig(input *go_hook.HookConfig) (config.HookConfig, er
...
-			return config.HookConfig{}, errors.New(`"Filterable" in KubernetesConfig cannot be nil`)
+		if kubeCfg.FilterFunc == nil {
+			return config.HookConfig{}, errors.New(`"FilterFunc" in KubernetesConfig cannot be nil`)
 		}
-		filterable := kubeCfg.Filterable
+		filterFunc := kubeCfg.FilterFunc
 		monitor.FilterFunc = func(obj *unstructured.Unstructured) (interface{}, error) {
-			return filterable.ApplyFilter(obj)
+			return filterFunc(obj)
 		}
```
[the remark this answers is not in the document]
Comment (zuzzas) [reply pkg/module_manager/global_hook_config.go:307]: These functions have different signatures.

[the remark this answers is not in the document]
Comment (diafour) [reply pkg/module_manager/global_hook_config.go:307]: Oh, it needs changes in shell-operator. OK.

**Review by nabokihms — APPROVED** [review]

File: sdk/registry.go (hunk 1 above)
Comment (zuzzas) [reply sdk/registry.go]: Removed all useless stubs and Register(). Only RegisterFunc() is left now since there is no practical need for a complete object.

**Review by diafour — APPROVED** [review]

**Pull request merged**